### ISSUE TITLE
Bug: Classifier V3 fails to categorize mixed-language (Hinglish) support tickets

### ISSUE THREAD

**Issue by harshitanagpal05** [riteshbonthalakoti/HELPDESK.AI#2418]
### Bug Description
The **Classifier V3** model (`backend/training/classifier_trainer_v3.py`) fails to accurately categorize support tickets containing **mixed-language input** (e.g., Hinglish — a blend of Hindi and English commonly used by Indian users). Tickets submitted in Hinglish are frequently mis-classified into incorrect categories or assigned wrong severity levels, resulting in poor ticket routing and delayed resolution.

The `MultiOutputClassifierV3` model uses a DistilBERT backbone fine-tuned primarily on English-language training data. When encountering code-mixed tokens (e.g., "mera laptop band ho gaya hai, please help"), the tokenizer produces excessive `[UNK]` tokens, degrading classification accuracy significantly.

---

### Steps to Reproduce
1. Start the backend server and ensure the Classifier V3 model is loaded.
2. Submit a new support ticket via the API with a mixed-language (Hinglish) description:
   ```json
   {
       "title": "Laptop issue",
       "description": "Mera laptop baar baar restart ho raha hai aur screen pe error aa raha hai, kya karu?"
   }
   ```
3. Observe the auto-assigned `category` and `severity` in the response payload.
4. Compare the classification against a purely English equivalent: _"My laptop keeps restarting and showing an error on screen, what should I do?"_
5. Notice the Hinglish ticket is frequently mis-categorized (e.g., assigned "General Inquiry" instead of "Technical Support") or given incorrect severity.

---

### Expected Behavior
The classifier should handle multilingual and code-mixed inputs gracefully by:
- Accurately classifying Hinglish tickets with at least 80% accuracy (on par with English-only inputs).
- Producing consistent category and severity assignments regardless of input language.
- Gracefully degrading with a confidence score when encountering unsupported language patterns, rather than silently mis-classifying.

---

### Root Cause Analysis & Solution
* **Root Cause**: The training dataset in `classifier_trainer_v3.py` is composed predominantly of English-language examples. The DistilBERT tokenizer (`distilbert-base-uncased`) is not optimized for Hindi or transliterated text, causing high `[UNK]` token rates for Romanized Hindi words.
* **Suggested Fix**:
  1. **Expand the training dataset** with multilingual edge-case examples, specifically Hinglish ticket samples covering common IT support scenarios.
  2. **Consider using a multilingual backbone** like `distilbert-base-multilingual-cased` or `xlm-roberta-base` to improve tokenization coverage for non-English scripts.
  3. **Add a pre-processing step** in the inference pipeline that detects code-mixed input and applies transliteration normalization before tokenization.
  4. **Expose a confidence score** in the classifier response so downstream systems can flag low-confidence classifications for human review via the retraining pipeline (`backend/training/retraining_pipeline.py`).

---

### Environment & Context
* **Affected Components**:
  * `backend/training/classifier_trainer_v3.py` (model architecture & training)
  * `backend/services/classifier_service.py` (inference service)
  * `backend/services/classifier_v3.py` (V3 service wrapper)
  * `backend/training/retraining_pipeline.py` (active learning loop)
* **Detected Sprint**: Sprint 6
* **Defect Log Reference**: `D-005` (Backend Team)
* **Status**: **Open** — Currently undergoing testing in the staging environment.
* **Tested Version / Branch**: `main` / `gssoc`

**Comment by riteshbonthalakoti** [comment]
Hi @harshitanagpal05! I have assigned this issue to you. Go ahead and start working! Make sure to sign up under the company **Ritesh PVT Limited** when testing. Let's build! 🚀
